pennylane/labs/vibrational_ham/localize_modes.py

```diff
@@ -0,0 +1,206 @@
+"""This module contains functions to localize normal modes."""
+
+import warnings
+import numpy as np
+import scipy
+
+au_to_cm = 219475
+hbar = 6.022 * 1.055e12  # (amu)*(angstrom^2/s)
+c_light = 3 * 10**8  # m/s
+
+# pylint: disable=dangerous-default-value, too-many-statements
+
+
+def _pm_cost(q):
+    r"""Pipek-Mezey cost function whose minimization yields localized displacements."""
+    nnuc, _, nmodes = q.shape
+
+    xi_pm = 0.0
+
+    for p in range(nmodes):
+        for i in range(nnuc):
+            q2 = 0.0
+            for alpha in range(3):
+                q2 += q[i, alpha, p] ** 2
+            xi_pm += q2**2
+
+    return -xi_pm
+
+
+def _mat_transform(u, qmat):
+    r"""Returns the rotated displacement vectors matrix for a given rotation unitary u and displacement vectors matrix qmat."""
+    qloc = np.einsum("qp,iaq->iap", u, qmat)
+
+    return qloc
+
+
+def _params_to_unitary(params, nmodes):
+    r"""Transforms a one-dimensional vector of parameters specifying a unitary rotation into its associated matrix u."""
+    ugen = np.zeros((nmodes, nmodes))
+
+    idx = 0
+    for m1 in range(nmodes):
+        for m2 in range(m1):
+            ugen[m1, m2] += params[idx]
+            ugen[m2, m1] -= params[idx]
+            idx += 1
+
+    return scipy.linalg.expm(ugen)
+
+
+def _params_cost(params, qmat, nmodes):
+    r"""Returns the cost function to be minimized for localized displacements."""
+    uparams = _params_to_unitary(params, nmodes)
+    qrot = _mat_transform(uparams, qmat)
+
+    return _pm_cost(qrot)
+
+
+def _q_normalizer(qmat):
```

Comment:
```suggestion
def _normalize_q(qmat):
```

Maybe this is better? `q_normalizer` sounds like a terrifying robot rather than a function somehow 😆